shared_memory_client.py

```diff
@@ -0,0 +1,170 @@
+            return response.response_code
+        else:
+            request = Request(self.SMT_GET_BY_COUNTER, [smt_object, counter, timeout, data_info_object])
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj, temp_data = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            self._copy_shared_memory_object(temp_data, data_info_object)
+            return response.response_code
+
+    def getByCounter(self, smt_object, counter, timeout, data_info_object=None):
+        if data_info_object is None:
+            request = Request(self.SMT_GET_BY_COUNTER, (smt_object, counter, timeout))
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            return response.response_code
+        else:
+            request = Request(self.SMT_GET_BY_COUNTER, (smt_object, counter, timeout, data_info_object))
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj, temp_data = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            self._copy_shared_memory_object(temp_data, data_info_object)
+            return response.response_code
+
+    def publish(self, smt_object):
+        request = Request(self.SMT_PUBLISH, smt_object)
+        self._connection.send(dumps(request))
+        response = loads(self._connection.receive(self.BUFFER_SIZE))
+        return response.response_code
+
+    def send(self, smt_object):
+        request = Request(self.SMT_PUBLISH, smt_object)
+        self._connection.send(dumps(request))
+        response = loads(self._connection.receive(self.BUFFER_SIZE))
+        return response.response_code
+
+    def latestRx(self, smt_object, data_info_object=None):
+        if data_info_object is None:
+            request = Request(self.SMT_GET_LATEST, smt_object)
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            return response.response_code
+        else:
+            request = Request(self.SMT_GET_LATEST, (smt_object, data_info_object))
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj, temp_data = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            self._copy_shared_memory_object(temp_data, data_info_object)
+            return response.response_code
+
+    def getLatest(self, smt_object, data_info_object=None):
```

Comment:
What is the difference between this and latestRx?

Reply:
same thing here as, two same functions, with an only different name.